src/engine/ScreenShotTaker.cs

```diff
@@ -26,45 +38,100 @@ public override void _Ready()
     public void TakeScreenshotPressed()
     {
         GD.Print("Taking a screenshot");
-        TakeAndSaveScreenShot();
+        TakeScreenshot();
     }
 
     /// <summary>
-    ///   Takes and saves a screenshot
+    ///   Takes an image of the current viewport
     /// </summary>
-    /// <returns>The path of the screenshot on success, null on failure</returns>
-    public string TakeAndSaveScreenShot()
+    /// <returns>The image</returns>
+    public Image GetViewportTextureAsImage()
+    {
+        var image = GetViewport().GetTexture().GetData();
+
+        // TODO: do we always need this?
+        image.FlipY();
+
+        return image;
+    }
+
+    private void SaveScreenshot(Image image)
     {
         FileHelpers.MakeSureDirectoryExists(Constants.SCREENSHOT_FOLDER);
 
-        var img = TakeScreenshot();
         var filename = DateTime.Now.ToString("yyyy-MM-dd_HH.mm.ss.ffff", CultureInfo.CurrentCulture) + ".png";
 
         var path = PathUtils.Join(Constants.SCREENSHOT_FOLDER, filename);
 
-        var error = img.SavePng(path);
+        var error = image.SavePng(path);
 
         if (error != Error.Ok)
         {
             GD.PrintErr("Saving screenshot failed ", error);
-            return null;
+            return;
         }
 
         GD.Print("Saved screenshot: ", path);
-        return path;
     }
 
     /// <summary>
-    ///   Takes an image of the current viewport
+    ///   Takes and saves a screenshot
     /// </summary>
-    /// <returns>The image</returns>
-    public Image TakeScreenshot()
+    private void TakeScreenshot()
     {
-        var image = GetViewport().GetTexture().GetData();
+        if (isCurrentlyTakingScreenshot)
+        {
+            GD.Print("Already in the process of taking a screenshot.");
+            return;
+        }
 
-        // TODO: do we always need this?
-        image.FlipY();
+        isCurrentlyTakingScreenshot = true;
 
-        return image;
+        saveScreenshotTask = new Task(() =>
+        {
+            SaveScreenshot(screenshotImage);
+            screenshotImage.Dispose();
+            screenshotImage = null;
+            isCurrentlyTakingScreenshot = false;
```

Comment:
I'd refactor this bit so that you pass in the image to this lambda, and the lambda body would call an instance method on this object, that way there isn't a need to have separate variables to hold this task or the screenshot image.